compiler/p/codegen/PPCSystemLinkage.cpp

```diff
@@ -641,13 +641,17 @@ TR_PPCSystemLinkage::createPrologue(
    cg()->setStackPointerRegister(machine->getPPCRealRegister(properties.getNormalStackPointerRegister()));
    TR::RealRegister        *sp = cg()->getStackPointerRegister();
    TR::RealRegister        *metaBase = cg()->getMethodMetaDataRegister();
+   TR::RealRegister        *gr2 = machine->getPPCRealRegister(TR::RealRegister::gr2);
    TR::RealRegister        *gr0 = machine->getPPCRealRegister(TR::RealRegister::gr0);
    TR::RealRegister        *gr11 = machine->getPPCRealRegister(TR::RealRegister::gr11);
    TR::RealRegister        *cr0 = machine->getPPCRealRegister(TR::RealRegister::cr0);
    TR::Node                   *firstNode = comp()->getStartTree()->getNode();
    TR::RealRegister::RegNum regIndex;
    int32_t                    size = bodySymbol->getLocalMappingCursor();
    int32_t                    argSize;
+    
+   //Move TOCBase into r2
+   cursor = loadConstant(cg(), firstNode, (int64_t)(cg()->getTOCBase()), gr2, cursor, false, false);
```

Comment:
You will need to guard this with `#ifdef __LITTLE_ENDIAN__` since this only applies to `ppc64le`.